apps/core-v2-testnet/scripts/deploy.ts

```diff
@@ -1,21 +1,66 @@
+import { BigNumber } from "ethers";
 import { ethers } from "hardhat";
-import Logger from "utils/logger";
+import {
+  deployForwarderFactory,
+  deployMockToken,
+  deployMockYearnVault,
+  deployMockYieldAdapter,
+} from "src/deploy";
+import { createTerm } from "src/helpers/createTerm";
+import { getCurrentBlockTimestamp } from "src/utils";
+import { ONE_DAY_IN_SECONDS, ONE_MINUTE_IN_SECONDS } from "@elementfi/base";
 
 async function main() {
-  const currentTimestampInSeconds = Math.round(Date.now() / 1000);
-  const ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60;
-  const unlockTime = currentTimestampInSeconds + ONE_YEAR_IN_SECS;
+  // get signers
+  const signers = await ethers.getSigners();
+  const [signer] = signers;
 
-  const lockedAmount = ethers.utils.parseEther("1");
+  // Deploy ForwarderFactory
+  const forwarderFactory = await deployForwarderFactory(signer);
 
-  const Lock = await ethers.getContractFactory("Lock");
+  // Deploy Mock Tokens
+  const usdcToken = await deployMockToken(signer, "USDC", "USDC", 6);
 
-  Logger.deployContract("Lock");
-  const lock = await Lock.deploy(unlockTime, { value: lockedAmount });
+  // ERC20Forwarder contract bytecode hash
+  const linkHash = await forwarderFactory.ERC20LINK_HASH();
 
-  await lock.deployed();
+  // deploy mock yearn vaults
+  // todo see how many decimals is usdc token
 
-  Logger.successfulDeploy("Lock", lock);
+  // Deploy mock vault
+  const mockYearnVault = await deployMockYearnVault(signer, usdcToken.address);
+
+  const governanceAddress = ethers.constants.AddressZero;
+  // Deploy mock yield adapter i.e. term contract
+  const mockYieldAdapter = await deployMockYieldAdapter(
+    signer,
+    mockYearnVault.address,
+    governanceAddress,
+    linkHash,
+    forwarderFactory.address,
+    usdcToken.address,
+  );
+
+  // give tokens to signer
+  usdcToken.connect(signer).mint(signer.address, 7e6);
+  // set allowance for yield adapter
+  usdcToken.approve(mockYieldAdapter.address, 12e6);
```

Comment:
Who's the spender? Update comment might be: "set allowance for yield adapter to spend signer's usdc"